feat: add firefox 148 spec

mingyech opened the pull request:
For browsers that sends pq hybrid and classical keyshare pairs, it may or may not choose to reuse the classical part between them. Add a helper to be used when crafting the spec to signify that the reuse should happen, and implement the reuse when generating keyshares. Existing specs will still generate keys independently.

Add Firefox 148 spec that uses this reuse.

mingyech commented:
Firefox is reusing the x25519 part for its X25519 and X25519MLKEM768 keyshares, different from the chrome behavior in #342

RPRX commented:
@mingyech 其实我觉得 Chrome 的做法有点浪费，但我猜是实现方式不同的原因，Chromium 里现在可能是每个密钥交换方式都对应着单独的函数，为未来某一天去掉 X25519 做准备

mingyech committed:
feat: add ff 148 spec and impl keyshare reuse

For browsers that sends pq hybrid and classical keyshare pairs, it may
or may not choose to reuse the classical part between them. Add a
helper to be used when crafting the spec to signify that the reuse
should happen, and implement the reuse when generating keyshares.
Existing specs will still generate keys independently.

Add Firefox 148 spec that uses this reuse.

mingyech committed:
fix: dont add unexported field

mingyech commented on u_public.go:
```diff
@@ -643,8 +643,17 @@ func (fh *finishedHash) getPublicObj() FinishedHash {
+// ApplyPreset reuses the same classical key material for both entries.
+func ReuseHybridAndClassicalKeyShares(hybrid, classical KeyShare) []KeyShare {
+	hybrid.hybridClassicalReuseWith = classical.Group
+	classical.hybridClassicalReuseWith = hybrid.Group
```
removed unexported field
